### ISSUE TITLE
Fix tests/test_executor.py in workers

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#2074]
https://github.com/huggingface/datasets-server/actions/runs/6802428610/job/18495505353

**Comment by albertvillanova** [comment]
But it seems that the test is not failing now...

**Comment by severo** [comment]
I understand that it fails randomly (mainly because we use a random sleep time in the code)

**Comment by albertvillanova** [comment]
The exit code was -9, so probably a SIGKILL?
```
E           mirakuru.exceptions.ProcessFinishedWithError: The process invoked by the <mirakuru.output.OutputExecutor: "/home/runner/.cache/pypoetry/virtualenvs/worker-ugd3P0FX-py3.9/bin/python /home/runner/work/datasets-server/datasets-server/services/worker/tests/test_executor.py --print-worker-state-path" 0x7f2008be6880> executor has exited with a non-zero code: -9.

/home/runner/.cache/pypoetry/virtualenvs/worker-ugd3P0FX-py3.9/lib/python3.9/site-packages/mirakuru/base.py:375: ProcessFinishedWithError
------------------------------ Captured log call -------------------------------
ERROR    root:executor.py:215 Worker crashed (exit code f-9 SIGKILL - surely an OOM)when running job_id=626173650000000000000000
```

**Comment by severo** [comment]
Trying to reproduce locally

```
cd services/worker
TEST_PATH=tests/test_executor.py make test
```

I ran it 5 times and had no error. I think we can close the issue indeed, and reopen if needed...

Related: should we try to replace

https://github.com/huggingface/datasets-server/blob/c21ea0779046aeb0ba09f52261f2d5c62f2419fe/services/worker/src/worker/executor.py#L107

with 

```python
        loop.add_signal_handler(signal.SIGKILL, self.stop)
        loop.add_signal_handler(signal.SIGTERM, self.stop)
```

to fix #2076? Not sure if it's related to this or not, and if SIGKILL is a signal that we can catch (I think not). But please give me your opinion if you have an idea.
